pmd-tsql/src/main/antlr4/net/sourceforge/pmd/lang/tsql/antlr4/TSqlLexer.g4

```diff
@@ -0,0 +1,1283 @@
+/*
+T-SQL (Transact-SQL, MSSQL) grammar.
+The MIT License (MIT).
+Copyright (c) 2017, Mark Adams (madams51703@gmail.com)
+Copyright (c) 2015-2017, Ivan Kochurkin (kvanttt@gmail.com), Positive Technologies.
+Copyright (c) 2016, Scott Ure (scott@redstormsoftware.com).
+Copyright (c) 2016, Rui Zhang (ruizhang.ccs@gmail.com).
+Copyright (c) 2016, Marcus Henriksson (kuseman80@gmail.com).
```

Comment:
From where did you take this grammar? Do you have a source? Is it somewhere on github? This might come in handy, if we need to update it.

Reply:
The lexer is coming from antlr4 grammars-v4 repository.
https://github.com/antlr/grammars-v4/tree/master/sql/tsql